### PR TITLE
Odx 284 dsm app hello world akeneo dsm front

### PR DESCRIPTION
**Pull request by QuentinMllt**

### CODE REVIEW

File: samples/dsm/akeneo-dsm/back-end/.env [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# Environment variables declared in this file are automatically made available to Prisma.
+# See the documentation for more detail: https://pris.ly/d/prisma-schema#accessing-environment-variables-from-the-schema
+
+# Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB and CockroachDB.
+# See the documentation for all the connection string options: https://pris.ly/d/connection-strings
+
+DATABASE_URL="file:/srv/app/var/nodejs_sample_apps.db"
+
+###> Application version ###
+APPLICATION_VERSION=1.1.0
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/.env:11]:
```suggestion
APPLICATION_VERSION=1.0.0
```

File: samples/dsm/akeneo-dsm/back-end/.env [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# This was inserted by `prisma init`:
+# Environment variables declared in this file are automatically made available to Prisma.
+# See the documentation for more detail: https://pris.ly/d/prisma-schema#accessing-environment-variables-from-the-schema
+
+# Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB and CockroachDB.
+# See the documentation for all the connection string options: https://pris.ly/d/connection-strings
+
+DATABASE_URL="file:/srv/app/var/nodejs_sample_apps.db"
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/.env:8]:
```suggestion
DATABASE_URL="file:/srv/app/var/dsm_apps.db"
```

File: samples/dsm/akeneo-dsm/back-end/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Start App Node JS + Express
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/README.md:1]:
```suggestion
# DSM App [README TODO]
```
Will be done in another ticket

File: samples/dsm/akeneo-dsm/back-end/tests/client/https-client.test.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+test("It constructs an HttpsClient object", async () => {
+
+    const pimUrl = new URL(process.env.AKENEO_PIM_URL);
+
+    expect(httpsClient.options.host).toBe(pimUrl.hostname);
+    expect(httpsClient.options.port).toBe(pimUrl.port);
+    expect(httpsClient.options.headers["User-Agent"]).toBe(
+        'AkeneoSampleApp/js-node-express'
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/tests/client/https-client.test.js:31]: I guess you have an error here when launching tests?

File: samples/dsm/akeneo-dsm/back-end/tests/client/https-client.test.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        + ' Version/' + process.env.APPLICATION_VERSION
+        + ' Docker/' + process.env.DOCKER_VERSION
+    );
+});
+
+test("It sets an acceptable userAgent", async () => {
+
+    let userAgent = httpsClient.buildUserAgent();
+    expect(userAgent).toBe('AkeneoSampleApp/js-node-express');
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/tests/client/https-client.test.js:40]: Here also?

File: samples/dsm/akeneo-dsm/back-end/package.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "js-node-express",
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/package.json:2]:
```suggestion
  "name": "dsm-app",
```

File: samples/dsm/akeneo-dsm/back-end/tests/use-cases/api/first-api-call.test.js
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/tests/use-cases/api/first-api-call.test.js:1]: To remove: no first api calll in dsm app

File: samples/dsm/akeneo-dsm/back-end/use-cases/api/index.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const { httpsClient } = require('../../client/index');
+const { tokenDb } = require('../../data-access/index');
+const LogicError = require("../../exceptions/logicError.exception");
+
+const doFirstApiCall = require('./first-api-call');
```
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/use-cases/api/index.js:5]: No more first API call

File: samples/dsm/akeneo-dsm/back-end/views/user.ejs
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/back-end/views/user.ejs:1]: Won't be used

File: samples/dsm/akeneo-dsm/front-end/README.md
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/README.md:1]: To be merge with to README of the DSM App (could be done to the README task later)

File: samples/dsm/akeneo-dsm/front-end/public/logo192.png
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/public/logo192.png:1]: to remove

File: samples/dsm/akeneo-dsm/front-end/public/logo512.png
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/public/logo512.png:1]: to remove

File: samples/dsm/akeneo-dsm/front-end/public/robots.txt
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/public/robots.txt:1]: I don't think we will use it ^^

File: samples/dsm/akeneo-dsm/front-end/src/logo.svg
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/src/logo.svg:1]: to remove

File: samples/dsm/akeneo-dsm/front-end/src/App.test.tsx
Comment (tomglvng) [inline samples/dsm/akeneo-dsm/front-end/src/App.test.tsx:1]: Front tests is a complex subject. I propose to not do them for the moment. What do you think?

File: samples/dsm/akeneo-dsm/front-end/README.md
Comment (QuentinMllt) [reply samples/dsm/akeneo-dsm/front-end/README.md:1]: Yes, I saw that there is a README task later so I will update it a this time.

File: samples/dsm/akeneo-dsm/front-end/src/App.test.tsx
Comment (QuentinMllt) [reply samples/dsm/akeneo-dsm/front-end/src/App.test.tsx:1]: Yes, I'm ok with that. I have never do that before.

File: samples/dsm/akeneo-dsm/back-end/.env (hunk 1 above)
Comment (QuentinMllt) [reply samples/dsm/akeneo-dsm/back-end/.env:11]: Ok, I update the version right now.

**Review by gwendalaubert — COMMENTED** [review]
OK except for an error with the permissions on node_modules due to a lack of user definition in docker-compose.yml.
Someone else should test to install it. I would suggest @JulienVerbrugge as he did not change his uid / gid to 1000.

**Pull request merged**